### PR TITLE
Allow user to dynamically set batch size of pixel SOM assignments

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

The current hard-coded batch size of 100 makes pixel SOM cluster assignment extremely slow. For a powerful machine (and even modern-day M1's and M2's), it can handle a much larger batch size and offer a 100x+ speedup.

**How did you implement your changes**

Pass an argument through the PixelSOMCluster process called `num_parallel_pixels`, which controls how many SOM clusters get assigned at once.

### CODE REVIEW

File: src/ark/phenotyping/cluster_helpers.py [hunk 1]
```diff
@@ -140,12 +147,12 @@ def generate_som_clusters(self, external_data: pd.DataFrame) -> np.ndarray:
 
         # work in batches of 100 to account to support large dataframe sizes
```
Comment (cliu72) [inline src/ark/phenotyping/cluster_helpers.py:148]: A verrrrry small comment - but it'd be good to update this comment here so it doesn't say "batches of 100"

**Review by cliu72 — CHANGES_REQUESTED** [review]
Sorry for the late review! Just some small requests.

**Review by cliu72 — CHANGES_REQUESTED** [review]
Can we add the same functionality to `assign_som_clusters` in the `CellSOMCluster` class in cluster_helpers.py (and cell_som_clustering.py respectively)?

**Comment by alex-l-kong** [comment]
> Can we add the same functionality to `assign_som_clusters` in the `CellSOMCluster` class in cluster_helpers.py (and cell_som_clustering.py respectively)?

Sure, same batch size work?

**Comment by cliu72** [comment]
> > Can we add the same functionality to `assign_som_clusters` in the `CellSOMCluster` class in cluster_helpers.py (and cell_som_clustering.py respectively)?
> 
> Sure, same batch size work?

Yup, that works.

**Review by cliu72 — APPROVED** [review]
Looks good!

**Pull request merged**